testcases/Test_build.py

```diff
@@ -0,0 +1,143 @@
+        self.append_kernel_cmdline = self.conf.args.append_kernel_cmdline
+        self.linux_path = os.path.join(self.home, "linux")
+        if not self.repo:
+            self.fail("Provide git repo of kernel to install")
+        if not (self.conf.args.host_ip and self.conf.args.host_user and self.conf.args.host_password):
+            self.fail(
+                "Provide host ip user details refer, --host-{ip,user,password}")
+
+    def build_kernel(self):
+        """
+        Build and install the Linux kernel.
+        """
+        
+        try:
+            
+            build_command = "make -j && make modules_install && make install"
+            err=self.connection.run_command(build_command, timeout=self.host_cmd_timeout)
+
+            
+            log.info("Kernel build successful")
+
+            return 0,err
+        except CommandFailed as e:
+            log.error("Kernel build failed: {}".format(e))
+            return  4,e
+    def generate_unique_id():
+            unique_id = str(uuid.uuid4())
+
+            return unique_id
+    def add_entry_to_json( self,data, entry):
+            entry_id = self.generate_unique_id()
+            data[entry_id] = entry
+            return data
+    def Store_loc ( self, er) :
+            pattern = r"([\w\d_]+\/(?:(?:[\w\d_]+\/)*[\w\d_]+\b))"
+            matches = [match.group(1) for match in re.finditer(pattern,er)]
+            return matches
+        
+    def runTest(self):
+        self.connection.run_command("if [ -d {} ]; then rm -rf {}; fi".format(self.home,self.home))
+        self.connection.run_command("if [ ! -d {} ]; then mkdir -p {}; fi".format(self.home,self.home))
+        self.connection.run_command("cd {}".format(self.home))
+        if not self.branch:
+            self.branch='master' 
+        log.info("CD DONE")
+        
+        self.connection.run_command("git clone --depth 1 {} -b {}".format(self.repo, self.branch))
+        self.connection.run_command("cd linux")
+        commit = self.connection.run_command(" git log -1 --format=%H")
+        self.connection.run_command("cd ..")
+        self.connection.run_command("wget http://ltc-jenkins.aus.stglabs.ibm.com:81/abdul/ioci/kernel_config -o linux/.config")
+        self.connection.run_command("cd linux && make olddefconfig")
+        error = self.build_kernel()
+        log.info("COMMAND RUN")
+        exit_code = error[0]
+        errVal = str(error[1])
+        log.info("printing the exit code '{}'".format(exit_code))
+        entry=[]
+        if exit_code != 0:
+            entry = self.Store_loc(errVal)[-1]
+            badCommit = commit[-1]
+            log.info("BUILD_BISECTOR CALLED")
+            bisect = Build_bisector.Buil_bisector()
+            bisect.setUp()
+            res = bisect.runTest()
+            log.info("BUILD_BISECTOR END")
+            em=res[0]
+            commiti=res[1]
+        else : 
+             em=""
+             commiti=commit[-1]
```

Comment:
please have meaningful  variable name which help us to well to maintain code

Reply:
changed